dashboards-observability/public/components/synthetics/helpers/test_details_flyout.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import _ from 'lodash';
+import React from 'react';
+import { FlyoutContainers } from '../../common/flyout_containers';
+import { Plt } from '../../visualizations/plotly/plot';
+import {
+  CONTENT_SIZE_FIELD,
+  DOWNLOAD_SPEED_FIELD,
+  DOWNLOAD_TIME_FIELD,
+  PRIMARY_IP_FIELD,
+  REDIRECT_TIMING_FIELD,
+  STARTTRANSFER_TIMING_FIELD,
+  TEST_SUITE_NAME_FIELD,
+  TIMING_FIELDS,
+  URL_FIELD,
+} from '../../../../common/constants/synthetics';
+
+interface TestDetailsFlyoutProps {
+  testSuiteObject: Record<string, any>;
+  closeFlyout: () => void;
+}
+
+export const TestDetailsFlyout = ({ testSuiteObject, closeFlyout }: TestDetailsFlyoutProps) => {
+  // contains all the timing fields being shown
+  const allTimingFields = () => {
+    let retval = [];
+    // slice is used to provide a shallow copy
+    retval = TIMING_FIELDS.slice();
+    retval.push(DOWNLOAD_TIME_FIELD);
+    retval.unshift('Content Download');
+    return retval;
+  };
+
+  // contains just the timing fields listed in TIMING_FIELDS
+  const getAllTimingValues = () => {
```

Comment:
i felt the naming is a bit confusing, `allTimingFields` includes `DOWNLOAD_TIME_FIELD` but `getAllTimingValues` does not? e.g. what does `all` mean?

Reply:
I think this is from timing responses from API calls. These include response time, time to first byte, etc.